src/main/java/net/imglib2/loops/LoopBuilder.java

```diff
@@ -78,68 +83,121 @@
-	{
-		this.images = images;
-		this.dimensions = new FinalInterval( images[ 0 ] );
-		Arrays.asList( images ).forEach( this::checkDimensions );
-	}
+	private MultiThreadSetting multiThreaded = MultiThreadSetting.SINGLE;
 
-	private void checkDimensions( final Interval interval )
-	{
-		final long[] a = Intervals.dimensionsAsLongArray( dimensions );
-		final long[] b = Intervals.dimensionsAsLongArray( interval );
-		if ( !Arrays.equals( a, b ) )
-			throw new IllegalArgumentException( "Dimensions do not fit." );
-	}
+	private boolean useFlatIterationOrder = false;
 
+	// public methods
+
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A > LoopBuilder< Consumer< A > > setImages( final RandomAccessibleInterval< A > a )
 	{
 		return new LoopBuilder<>( a );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A, B > LoopBuilder< BiConsumer< A, B > > setImages( final RandomAccessibleInterval< A > a, final RandomAccessibleInterval< B > b )
 	{
 		return new LoopBuilder<>( a, b );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A, B, C > LoopBuilder< TriConsumer< A, B, C > > setImages( final RandomAccessibleInterval< A > a, final RandomAccessibleInterval< B > b, final RandomAccessibleInterval< C > c )
 	{
 		return new LoopBuilder<>( a, b, c );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A, B, C, D > LoopBuilder< FourConsumer< A, B, C, D > > setImages( final RandomAccessibleInterval< A > a, final RandomAccessibleInterval< B > b, final RandomAccessibleInterval< C > c, final RandomAccessibleInterval< D > d )
 	{
 		return new LoopBuilder<>( a, b, c, d );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A, B, C, D, E > LoopBuilder< FiveConsumer< A, B, C, D, E > > setImages( final RandomAccessibleInterval< A > a, final RandomAccessibleInterval< B > b, final RandomAccessibleInterval< C > c, final RandomAccessibleInterval< D > d, final RandomAccessibleInterval< E > e )
 	{
 		return new LoopBuilder<>( a, b, c, d, e );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public static < A, B, C, D, E, F > LoopBuilder< SixConsumer< A, B, C, D, E, F > > setImages( final RandomAccessibleInterval< A > a, final RandomAccessibleInterval< B > b, final RandomAccessibleInterval< C > c, final RandomAccessibleInterval< D > d, final RandomAccessibleInterval< E > e, final RandomAccessibleInterval< F > f )
 	{
 		return new LoopBuilder<>( a, b, c, d, e, f );
 	}
 
+	/**
+	 * @see LoopBuilder
+	 */
 	public void forEachPixel( final T action )
 	{
 		Objects.requireNonNull( action );
-		final List< RandomAccess< ? > > samplers = Stream.of( images ).map( this::initRandomAccess ).collect( Collectors.toList() );
-		final Positionable synced = SyncedPositionables.create( samplers );
-		LoopUtils.createIntervalLoop( synced, dimensions, RunnableFactory.bindActionToSamplers( action, samplers ) ).run();
+		if ( Intervals.numElements( dimensions ) == 0 )
+			return;
+		if ( allAreNativeImages() )
```

Comment:
This is not a particular good criterion to decide for/against Cursors.
Views can have very fast Cursors, for example if you do a few transformations on an ArrayImg that concatenate to identity, you will get a ArrayImg Cursor (although your input image is a IntervalView or something like that).
It would be better to look at the Cursor types.

Reply:
I tried, but this is not an easy problem: How do I know if a cursor is fast?